### PR TITLE
Support storage type selection while vault creation

### PR DESCRIPTION
**Pull request by nilaydshah**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [X] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*Support storage type selection while vault creation

### CODE REVIEW

**Comment by nilaydshah** [comment]
@bmoore-msft can you please review this PR? (Updated existing template instead of creating new one as per PR https://github.com/Azure/azure-quickstart-templates/pull/5467 )

File: 101-recovery-services-vault-create/azuredeploy.json [hunk 1]
```diff
@@ -26,17 +34,35 @@
...
-    }
+    },
+    {
+      "condition": "[parameters('changeStorageType')]",
+      "type": "Microsoft.RecoveryServices/vaults/backupstorageconfig",
+      "name": "[concat(parameters('vaultName'), '/vaultstorageconfig')]",
+      "apiVersion": "2018-01-10",
+      "properties": {
+          "StorageModelType":"[parameters('vaultStorageType')]",
+          "StorageType":null,
```
Comment (bmoore-msft) [inline 101-recovery-services-vault-create/azuredeploy.json]: remove null properties

File: 101-recovery-services-vault-create/azuredeploy.json [hunk 2]
```diff
@@ -26,17 +34,35 @@
...
+          "StorageTypeState":null
+      },
+      "dependsOn": [
```
Comment (bmoore-msft) [inline 101-recovery-services-vault-create/azuredeploy.json]: move this above properties

File: 101-recovery-services-vault-create/README.md [hunk 3]
```diff
@@ -14,3 +14,6 @@ A disaster recovery and data protection strategy keeps your business running whe
...
 
 The Site Recovery service ensures your servers, virtual machines, and apps are resilient by replicating them so that when disasters and outages occur you can easily fail over to your replicated environment and continue working. When services are resumed you simply failback to your primary location with uninterrupted access. Site Recovery helps protect a wide range of Microsoft and third-party workloads. [Learn more] (http://aka.ms/asr-learn-more/).
+
+#### Storage Type Selection
+A Recovery Services vault can only change storage options before any backups have been stored. Once any backup is configured, storage type cannot be changed. Hence use conditional parameter to opt-in or opt-out change storage type [Learn more] (https://docs.microsoft.com/en-us/azure/backup/backup-azure-backup-faq#can-i-change-from-grs-to-lrs-after-a-backup)
```
Comment (bmoore-msft) [inline 101-recovery-services-vault-create/README.md]: fix grammar:
… the storage type …
… use the conditional parameter …

File: 101-recovery-services-vault-create/azuredeploy.json [hunk 4]
```diff
@@ -8,14 +8,22 @@
         "description": "Name of the Vault"
       }
     },
-    "skuTier": {
+    "changeStorageType": {
+      "type": "bool",
+      "defaultValue": false,
+      "metadata": {
+        "description": "Change Vault Storage Type Selection (Works if vault has not registered any backup instance)"
```
Comment (bmoore-msft) [inline 101-recovery-services-vault-create/azuredeploy.json]: Change Vault Storage Type (not allowed if the vault has registered backups)

File: 101-recovery-services-vault-create/azuredeploy.parameters.json [hunk 5]
```diff
@@ -5,8 +5,11 @@
         "vaultName": {
             "value": "RSvault1"
         },
-        "skuTier": {
-            "value": "Standard"
+        "changeStorageType": {
```
Comment (bmoore-msft) [inline 101-recovery-services-vault-create/azuredeploy.parameters.json]: remove the params that have defaultValues

**Comment by nilaydshah** [comment]
@bmoore-msft I've fixed all review comments. Can you please take a look?

**Comment by nilaydshah** [comment]
@bmoore-msft I've fixed all review comments. Can you take a look?

**Pull request merged**